src/plugins/intel_gpu/include/intel_gpu/primitives/reorder.hpp

```diff
@@ -24,6 +24,13 @@ enum class reorder_mean_mode {
     div,       // val/mean
 };
 
+/// @brief channel operation modes
+enum class channel_mode {
```

Comment:
[random spot] The overall solution looks like a big workaround. I suggest reworking it.
Since reorder kernel doesn't support nv12->grayscale conversion and adding it requires bunch of hacks, I suggest using `convert_color` primitive instead. You can add `color_format::GRAY` mode and implement corresponding impl in ocl kernel, and then just create `convert_color` instead of `reorder` for such cases.

Another concern is that the solution is implemented for API 1.0 only... Looks like we'll need to add `NV12ToGray` operation for such cases and add corresponding changes in `Parameter` factory in gpu plugin